### PR TITLE
Add factories for model users.User

### PR DESCRIPTION
**Pull request by ad-m**

### CODE REVIEW

**Comment by pydanny** [comment]
This functionality is encapsulated by django-test-plus' `self.make_user()` function. Which means one of the following:

* We need to document why we don't include a `UserFactory`
* There might be a good reason to include this factory.

@frankwiles, @ad-m Do you have any comments or thoughts?

**Comment by ad-m** [comment]
I agree that UserFactory alone can be replaced by ```self.make_user()```, but cookiecutter-django is starter for other projects, where ```UserFactory``` is powerfull to use as subfactory (see ```factory.SubFactory``` from factoryboy). Using ```self.make_user()``` to populate subfactories (or (sub)^x factories) sounds awful.

**Comment by ad-m** [comment]
I added default password for ```UserFactory``` like ```self.make_user()```, so you can log-in freely by ```self.client.login(username=UserFactory().username, password='password')``` too.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/users/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from feder.users import models
```
Comment (Eraldo) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/users/tests/factories.py]: Just out of curiosity..
Where does `feder` come from?

**Comment by frankwiles** [comment]
I agree with @ad-m ```self.make_user()``` is fine for simple tests, but having a true User factory is more powerful, but often project specific.  In my experience, if you're making a custom user model it's very likely for things like removing username and using email login only or other major surgery that ```make_user()``` isn't ever going to be able to support.

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/users/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import factory
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    username = factory.Sequence(lambda n: 'user-{0}'.format(n))
+    email = factory.Sequence(lambda n: 'user-{0}@example.com'.format(n))
+    password = factory.PosteGnerationMethodCall('set_password', 'password')
```
Comment (pydanny) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/users/tests/factories.py]: `PosteGnerationMethodCall` appears to be a typo. :wink:

**Comment by pydanny** [comment]
@ad-m Great explanation! @frankwiles, thank you for your commentary. :smile: 

@ad-m Can you fix the typos so we can accept this pull request? :sunglasses:

**Pull request merged**